plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoClientConfig.java

```diff
@@ -59,6 +60,19 @@
     private String requiredReplicaSetName;
     private String implicitRowFieldPrefix = "_pos";
 
+    @AssertTrue(message = "Exactly one of these 'mongodb.seed' or 'mongodb.connection-url' must be specified")
+    public boolean isConnectionPropertyValid()
+    {
+        // Also needs to check if the existing value is empty.
+        // isPresent on connectionUrl default Optional.empty() returns false
+        // isPresent on test value empty string returns true
+        if (seeds.isEmpty() && (connectionUrl.isPresent() && connectionUrl.get().isEmpty())) {
+            return false;
+        }
+
+        return seeds.isEmpty() || (connectionUrl.isPresent() && connectionUrl.get().isEmpty());
```

Comment:
What we want to make sure here is which property is configured and the value validation of `mongodb.connection-url` isn't mandatory. Furthermore, empty string check is insufficient. I would recommend adding `@Pattern` annotation to getter of connection-url instead.

Reply:
@Pattern doesn't work on optionals

Reply:
Try `Optional<@Pattern...`.

Reply:
That doesn't work

Reply:
Replied to your DM in Slack.